spring-integration-gcp/src/main/java/org/springframework/integration/gcp/outbound/PubSubMessageHandler.java

```diff
@@ -29,15 +37,28 @@
 
 	private final PubSubOperations pubSubTemplate;
 
+	private MessageConverter messageConverter;
+
 	private String topic;
 
 	public PubSubMessageHandler(PubSubOperations pubSubTemplate, String topic) {
 		this.pubSubTemplate = pubSubTemplate;
 		this.topic = topic;
+
+		StringMessageConverter stringMessageConverter = new StringMessageConverter();
+		stringMessageConverter.setSerializedPayloadClass(String.class);
+		this.messageConverter =
+				new CompositeMessageConverter(ImmutableList.of(stringMessageConverter));
 	}
 
 	@Override
-	protected void handleMessageInternal(Message<?> message) throws Exception {
-		this.pubSubTemplate.publish(this.topic, message);
+	protected void handleMessageInternal(Message<?> message) {
+		String payload = (String) this.messageConverter.fromMessage(message, String.class);
```

Comment:
Pub/Sub represents the message payload as `ByteString`. So, I think we should be converting directly to `ByteString` here instead of `String`.

Reply:
In addition I would like to allow to send messages to this `PubSubMessageHandler` with the payload as `PubsubMessage`, `ByteString`, `byte[]` bypassing any conversion at all.

Reply:
@artembilan you already have `PubSubTemplate` for that. Why should that be in the `PubSubMessageHandler`? `AbstractMessageHandler` only declares `handleMessageInternal(Message)`.

Reply:
Well, I have an Integration flow. All the components are connected via message channels which are agnostic of the `payload` of the message they transport. And you can't predict what upstream logic I have. So, why would someone force me to switch to the `PubSubTemplate` if I would like to send message to the channel and don't care. The target `MH` must be smart and let me to send different `payload`s.

I don't talk about `Message`, @joaoandremartins , I talk about its `payload`. 
That is one of the gain of Spring Integration - let end-user do his logic in his own domain and be as much flexible for his `payload` as possible.

I feel that you are mixing somehow purpose of Channel Adapter and Template. That's why I suggested in the beginning do not look to the Template abstraction and just concentrate on the comprehensive Channel Adapters implementations.

Sorry if I'm confusing you but right now it is somehow in a mess.

Reply:
Apologies, I didn't understand what you meant at first.
Yes, I agree we must be able to send `Messages` with payloads other than String.